Mg/md

mgastegger opened the pull request:
MD code for new SchNetPack.

ktschuett commented on src/schnetpack/md/md_configs/dynamics/barostat/nhc_aniso.yaml:
```diff
@@ -0,0 +1,10 @@
+_target_: nhc_barostat_aniso
```
use exlicit clases here

mgastegger replied:
removed alias

ktschuett commented on src/schnetpack/md/md_configs/calculator/schnetpack.yaml:
```diff
@@ -0,0 +1,14 @@
+_target_: schnetpack
```
not a good alias

mgastegger replied:
Completely removed alias

mgastegger commented on src/schnetpack/md/simulation_hooks/barostats.py:
```diff
@@ -0,0 +1,904 @@
+            + system.forces * system.expand_atoms(b_coeff) * self.time_step * 0.5
+        )
+
+    # def compute_conserved(self, system):
```
I would prefer to leave this in for future debugging. This computes the conserved quantity of the thermostat to check whether it is implemented correctly. It is not necessary for production runs.

mgastegger commented on src/schnetpack/md/utils/base.py:
```diff
@@ -0,0 +1,33 @@
+log = logging.getLogger(__name__)
+
+
+def set_random_seed(seed):
```
Deleted function and module. The MD was already using `pl.seed_everything`.

ktschuett commented on src/schnetpack/md/simulator.py:
```diff
@@ -0,0 +1,261 @@
+
+        with grad_context:
+            # Perform initial computation of forces
+            self.calculator.calculate(self.system)
```
`self.system` could turn out to be a problem...

mgastegger replied:
Keep as is for now

mgastegger commented on src/schnetpack/md/md_configs/calculator/schnetpack_ensemble.yaml:
```diff
@@ -0,0 +1,15 @@
+_target_: schnetpack_ensemble
```
Removed alias

mgastegger commented:
- Completely removed alias functions
- Restructured configs
- Added pytorch 1.9.0 lazy init functionality for registering buffers properly
- Fixed restarts
- Added logging of pressure, potential energy and stress to HDF5 and added read to loader

ktschuett commented on src/schnetpack/md/calculators/schnetpack_calculator.py:
```diff
@@ -0,0 +1,240 @@
+__all__ = ["SchnetPackCalculator", "SchnetPackEnsembleCalculator"]
+
+
+class SchnetPackCalculator(MDCalculator):
```
SchNetPack with capital N :)

ktschuett commented on src/schnetpack/atomistic/model.py:
```diff
@@ -95,10 +95,13 @@ def __init__(
+                for p in m.required_derivatives:
+                    if p not in self.required_derivatives:
+                        self.required_derivatives.append(p)
+
```
couldn't you keep it a set and then do: `self.required_derivatives=list(required_derivatives)`?

mgastegger replied:
This was a leftover from the torch script trial. Reverted it to original set version

ktschuett commented on src/schnetpack/md/calculators/schnetpack_calculator.py:
```diff
@@ -0,0 +1,240 @@
+        return inputs
+
+
+class SchnetPackEnsembleCalculator(EnsembleCalculator, SchnetPackCalculator):
```
N!
